### PR TITLE
Add Projects view - list/table view [SCI-2731]

### PR DESCRIPTION
**Pull request by okriuchykhin**
https://biosistemika.atlassian.net/browse/SCI-2731

### CODE REVIEW

File: app/assets/javascripts/projects/index.js [hunk 1]
```diff
@@ -384,6 +439,218 @@
...
+        // Get row ID
+        var rowId = data.DT_RowId;
+        var dropdown = $(row).find('.dropdown');
+        var dropdownCell = dropdown.closest('td');
+        // If row ID is in the list of selected row IDs
+        if ($.inArray(rowId, selectedProjects) !== -1) {
+          $(row).find('input[type="checkbox"]').prop('checked', true);
+        }
+
+        initEditProjectButton($(row));
```
Comment (Ducz0r) [inline app/assets/javascripts/projects/index.js]: This `rowCallback` is triggered any time table is reloaded, e.g. on sorting of columns, filtering, etc. And in this function, we're initializing various event listeners - for edit & archive/restore functionalities. But these event listeners never get de-registered, do they? Could we somehow de-register them before (is there some "on table unload" callback) reloading the table?

File: config/routes.rb [hunk 2]
```diff
@@ -188,7 +188,7 @@ def draw(routes_name)
     end
 
     get 'projects/archive', to: 'projects#archive', as: 'projects_archive'
-    get 'projects/index_dt', to: 'projects#index_dt', as: 'projects_index_dt'
+    post 'projects/index_dt', to: 'projects#index_dt', as: 'projects_index_dt'
```
[the remark this answers is not in the document]
Comment (okriuchykhin) [reply config/routes.rb]: It is probably for better parameter handling, as it sends quite complex hash of parameters

File: app/assets/javascripts/projects/index.js (hunk 1 above)
Comment (okriuchykhin) [reply app/assets/javascripts/projects/index.js]: I think they should be removed together with DOM elements, they attached to, once content of the table is re-drawn.

File: app/views/projects/index/_project.html.erb [hunk 3]
```diff
@@ -5,41 +5,7 @@
...
-              <a href="#"
-                class="form-submit-link"
-                data-turbolinks="false"
-                data-submit-form="<%= project_form.options[:html][:id] %>"><%= t 'projects.index.restore_option' %></a>
-            </li>
-          <% end %>
-        </ul>
-      </div>
-    <% end %>
+    <%= render partial: "projects/index/project_actions_dropdown.html.erb", locals: { project: project, view_mode: 'cards' } %>
```
[the remark this answers is not in the document]
Comment (okriuchykhin) [reply app/views/projects/index/_project.html.erb]: I just forgot to remove it

[the remark this answers is not in the document]
Comment (okriuchykhin) [reply app/views/projects/index/_project.html.erb]: Removed

**Comment by Ducz0r** [comment]
Alright, good job! You can :ship: away.

**Pull request merged**